discovery/kubernetes/kubernetes.go

```diff
@@ -804,3 +804,19 @@ func addObjectMetaLabels(labelSet model.LabelSet, objectMeta metav1.ObjectMeta,
 func namespacedName(namespace, name string) string {
 	return namespace + "/" + name
 }
+
+// deletedNodeName knows how to handle the cache.DeletedFinalStateUnknown tombstone.
+func deletedNodeName(o interface{}) (string, error) {
+	node, ok := o.(*apiv1.Node)
+	if !ok {
+		tombstone, ok := o.(cache.DeletedFinalStateUnknown)
```

Comment:
IIUC when o is a `cache.DeletedFinalStateUnknown` , tombstone.Key is the name of the node so we shouldn't have to error out.

Reply:
You mean at `node, ok = tombstone.Obj.(*apiv1.Node)`? same as https://github.com/prometheus/prometheus/pull/15170#discussion_r1802501377
If we realize it's not sufficient, we can always "hardcode" the keyFunc on our side and just rely on `.Key`, I'll add a comment for that.